rules/aar_import/impl.bzl

```diff
@@ -351,7 +351,16 @@ def _process_jars(
         # TODO(djwhang): AarImportTest is not expecting jdeps, enable or remove it completely
         # jdeps = jdeps_artifact,
     )
-    providers.append(java_info)
+
+    # Merge library java_info with the resource java_info
+    # Libraries that directly depend on an AAR can depend as well on the AAR resource class
+    # so it is not enough to simply return the resource info as a dependency.
+    # Note that adding r_java as an export is also not the correct behavior, since we only
+    # want to expose it as a direct dependency.
+    # see AarImport.java
+    # https://github.com/bazelbuild/bazel/blob/e0a9081f/src/main/java/com/google/devtools/build/lib/rules/android/AarImport.java#L166
+    combined_java_info = java_common.merge([java_info] + r_java_info)
+    providers.append(combined_java_info)
```

Comment:
This is currently breaking some assumptions in internal tests. I will try to follow up on whether we should change the assumptions or not.

Reply:
I tested this some more: if you replace line 349's `_acls.in_aar_import_exports_r_java(str(ctx.label))` with True and undo your change from lines 354-364, it makes internal tests pass again.  Does that behavior work for your usecase? Merging the r_java_info with the java_info causes internal assumptions about aar_import providers to break (i.e. internally we are expecting only one srcjar to come out of aar_import, whereas your PR seems to generate two), which makes accepting this PR much more difficult.